library/classes/Installer.class.php

```diff
@@ -473,304 +483,329 @@ public function write_configuration_file() {
+            if (! empty($this->clone_database)) {
+                if ( ! $this->create_dumpfiles() ) {
+                    return False;
+                }
+            }
+            // Create the site directory
+            //   (applicable if mirroring another local site)
+            if ( ! empty($this->source_site_id) ) {
+                if ( ! $this->create_site_directory() ) {
+                    return False;
+                }
+            }
+            $this->disconnect();
+            if (! $this->user_database_connection()) {
+                // Re-connect to mysql via root user
+                if (! $this->root_database_connection() ) {
+                    return False;
+                }
+                // Create the mysql database
+                if ( ! $this->create_database()) {
+                    return False;
+                }
+                // Grant user privileges to the mysql database
+                if ( ! $this->grant_privileges() ) {
+                    return False;
+                }
+            }
+            $this->disconnect();
+        }
+        // Connect to mysql via created user
+        if ( ! $this->user_database_connection() ) {
+            return False;
+        }
+        // Build the database
+        if ( ! $this->load_dumpfiles() ) {
+            return False;
+        }
+        // Write the sql configuration file
+        if ( ! $this->write_configuration_file() ) {
+            return False;
+        }
+        // Load the version information, globals settings,
+        // initial user, and set up gacl access controls.
+        //  (applicable if not cloning from another database)
+        if (empty($this->clone_database)) {
+            if ( ! $this->add_version_info() ) {
+                return False;
+            }
+            if ( ! $this->insert_globals() ) {
+                return False;
+            }
+            if ( ! $this->add_initial_user() ) {
+                return False;
+            }
+            if ( ! $this->install_gacl()) {
+                return False;
+            }
+        }
+
+        return True;
     }
 
-    $install_results_2 = $this->get_require_contents($this->gaclSetupScript2);
-    if (! $install_results_2 ) {
-      $this->error_message = "install_gacl failed: unable to require gacl script 2";
-      return FALSE;
+    private function execute_sql( $sql, array $value = null , $str='', $count=0) {
+
+        $this->error_message = '';
+        if ( ! $this->dbh ) {
+            $this->user_database_connection();
+        }
+        if($count == 0){
+            $results = $this->dbh->query($sql);
+        }else{
+            $stmt = $this->dbh->prepare($sql);
+            if($count == 1){
+                $stmt->bind_param($str,$value[0]);
+            }
+            if($count == 2){
+                $stmt->bind_param($str,$value[0],$value[1]);
+            }
+            if($count == 3){
```

Comment:
No this is baaaaaad. Try another approach how about escaping the query then binding the array in here. Try another alternative.

Reply:
What if they were 20 params

Reply:
had issues with knowing how many elements the array has to index it and as I saw the number number coming from the user did not exceed 5...

Reply:
number of inputs*

Reply:
php > 5 doesn't support variable decomposition also

Reply:
if you have any clue...I will gladly like to hear

Reply:
This is where you communicate and ask questions.

Reply:
@robbyoconnor  thanks